install.ps1

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env pwsh
+# Copyright 2024 Science project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+<#
+.SYNOPSIS
+  Installs the `science.exe` executable.
+
+.DESCRIPTION
+  Downloads and installs the latest version of the science executable by default.
+  The download is first verified against the published checksum before being installed.
+  The install process will add the science executable to your user PATH environment variable
+  if needed as well as to the current shell session PATH for immediate use.
+
+.PARAMETER Help
+  Display this help message.
+
+.PARAMETER BinDir
+  The directory to install the science binary in.
+
+.PARAMETER NoModifyPath
+  Do not automatically add -BinDir to the PATH.
+
+.PARAMETER Version
+  The version of the science binary to install, the latest version by default.
+  The available versions can be seen at:
+    https://github.com/a-scie/lift/releases
+
+.INPUTS
+  None
+
+.OUTPUTS
+  The path of the installed science executable.
+
+.LINK
+  Docs https://science.scie.app
+
+.LINK
+  Chat https://scie.app/discord
+
+.LINK
+  Source https://github.com/a-scie/lift
+#>
+
+param (
+  [Alias("h")]
+  [switch]$Help,
+
+  [Alias("d")]
+  [string]$BinDir = (
+    # N.B.: PowerShell>=6 supports varargs, but to retain compatibility with older PowerShell, we
+    # just Join-Path twice.
+    # See: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/join-path?view=powershell-7.4#-additionalchildpath
+    Join-Path (
+      Join-Path ([Environment]::GetFolderPath("LocalApplicationData")) "Programs") "Science"
+  ),
+
+  [switch]$NoModifyPath,
+
+  [Alias("V")]
+  [string]$Version = "latest/download"
+)
+
+$ErrorActionPreference = 'Stop'
+
+function Green {
+  [Parameter(Position=0)]
+  param ($Message)
+
+  Write-Host $Message -ForegroundColor Green
+}
+
+function Warn {
+  [Parameter(Position=0)]
+  param ($Message)
+
+  Write-Host $Message -ForegroundColor Yellow
+}
+
+function Die {
+  [Parameter(Position=0)]
+  param ($Message)
+
+  Write-Host $Message -ForegroundColor Red
+  exit 1
+}
+
+function TemporaryDirectory {
+  $Tmp = [System.IO.Path]::GetTempPath()
+  $Unique = (New-Guid).ToString("N")
+  $TempDir = New-Item -ItemType Directory -Path (Join-Path $Tmp "$Unique")
+  trap {
+    Remove-Item $TempDir -Recurse
+  }
+  return $TempDir
+}
+
+function Fetch {
+  param (
+    [string]$Url,
+    [string]$DestFile
+  )
+
+  Invoke-RestMethod -Uri $Url -OutFile $DestFile
```

Comment:
So you can Pin SSL protocol versions with `-SslProtocol Tls12,Tls13`, but the option is only supported in PowerShell>=6 (its at 7.4.x now) and Tls13 is only available for PowerShell>=7.1. Windows 11 still ships with PowerShell 5.1.x by default. As a result I just went with the lowest common denominator here.

Reply:
I went ahead and added the pinning.